src/types.d.ts

```diff
@@ -106,6 +110,19 @@ declare module '@google-cloud/common' {
   }
 }
 
+// There is a @types/extend, but it's not expressive enough.
```

Comment:
Can we submit this as a PR?

Reply:
Yep -- but while our attention is focused on it, what do you think of dropping `extend` for `Object.assign`? Since we already target Node 4+, which is the criteria for using the latter.

Reply:
sgtm.

Reply:
Done -- changed to reference the PR (https://github.com/DefinitelyTyped/DefinitelyTyped/pull/20266)

Reply:
I'll remove the `extend` dependency in a future PR.